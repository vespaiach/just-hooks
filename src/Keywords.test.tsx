import React from 'react';
import { beforeEach, afterEach, test, expect, vi } from 'vitest';
import { create, ReactTestRenderer, ReactTestRendererJSON } from 'react-test-renderer';
import { unmountComponentAtNode } from 'react-dom';
import KeywordList, { Keyword } from './Keywords';

let container: HTMLDivElement | null = null;
beforeEach(async () => {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    if (container) {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    }
});

test('Keyword component snapshot', () => {
    const component = create(
        <Keyword name="react" url="https://reactjs.org/" className="keyword" style={{ fontWeight: 'bold' }}>
            <span>(111)</span>
        </Keyword>,
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
});

test('Keyword List component snapshot', () => {
    const keywords = [
        { name: 'react', url: 'https://reactjs.org/', extra: <span>111</span> },
        { name: 'vue', url: 'https://vuejs.org/', extra: <span>111</span> },
    ];
    const component = create(
        <KeywordList className="keyword-list" style={{ fontWeight: 'bold' }} keywords={keywords} as="p">
            <span>(111 items)</span>
        </KeywordList>,
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
});

function toJson(component: ReactTestRenderer) {
    const result = component.toJSON();
    expect(result).toBeDefined();
    return result as ReactTestRendererJSON;
}
