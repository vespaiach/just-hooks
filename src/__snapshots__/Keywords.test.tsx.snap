// Vitest Snapshot v1

exports[`Keyword List component snapshot 1`] = `
<p
  className="keyword-list"
  itemProp="keywords"
  itemScope={true}
  itemType="https://schema.org/DefinedTerm"
  style={
    {
      "fontWeight": "bold",
    }
  }
>
  <a
    href="https://reactjs.org/"
    itemProp="url"
    title="react"
  >
    <span
      itemProp="name"
    >
      react
    </span>
    <span>
      111
    </span>
  </a>
  <a
    href="https://vuejs.org/"
    itemProp="url"
    title="vue"
  >
    <span
      itemProp="name"
    >
      vue
    </span>
    <span>
      111
    </span>
  </a>
  <span>
    (111 items)
  </span>
</p>
`;

exports[`Keyword component snapshot 1`] = `
<a
  className="keyword"
  itemProp="url"
  style={
    {
      "fontWeight": "bold",
    }
  }
>
  <span
    itemProp="name"
  >
    react
  </span>
  <span>
    (111)
  </span>
</a>
`;
